codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -163,7 +163,11 @@ public ChannelFuture writeHeaders(final ChannelHandlerContext ctx, final int str
             Http2Stream stream = connection.stream(streamId);
             if (stream == null) {
                 try {
-                    stream = connection.local().createStream(streamId, endOfStream);
+                    // We don't create the stream in a `halfClosed` state because if this is an initial
+                    // HEADERS frame we don't want to connection state to transition until after the
```

Comment:
`we don't want to connnection state to transition` - Perhaps you meant `we don't want connection state to enter 'halfClosed' state`?

Reply:
The comment was pretty poor in general. I attempted to clarify it, both in terms of wording and the intent, specifically that we don't want the `stream.isHeadersSent()` to be `true` until we've actually sent the headers.